File: udp_client.go
/*
 * UDP Client in Go
 * Sends a message to UDP ports 8080, 8081, 8082, and 8083 on localhost
 *
 * Usage:
 *     ./udp_client_go "Your message here"
 *
 * How to compile:
 *     go build -o udp_client_go udp_client.go
 *
 * How to run:
 *     ./udp_client_go "Hello, World!"
 */

package main

import (
	"fmt"
	"net"
	"os"
)

func main() {
	if len(os.Args) != 2 {
		fmt.Printf("Usage: %s \"Your message here\"\n", os.Args[0])
		os.Exit(1)
	}

	message := os.Args[1]

	// Ports to send the message to
	ports := []int{8080, 8081, 8082, 8083}

	// Send the message to each port
	for _, port := range ports {
		addr := fmt.Sprintf("127.0.0.1:%d", port)
		conn, err := net.Dial("udp", addr)
		if err != nil {
			fmt.Printf("Failed to connect to %s: %v\n", addr, err)
			continue
		}

		_, err = conn.Write([]byte(message))
		if err != nil {
			fmt.Printf("Failed to send to %s: %v\n", addr, err)
		} else {
			fmt.Printf("Message sent to port %d\n", port)
		}

		conn.Close()
	}
}

File: udp_client.rs
// UDP Client in Rust
// Sends a message to UDP ports 8080, 8081, 8082, and 8083 on localhost
//
// Usage:
//     ./udp_client "Your message here"
//
// How to compile:
//     rustc -o udp_client_rs udp_client.rs
//
// How to run:
//     ./udp_client_rs "Hello, World!"

use std::env;
use std::net::UdpSocket;

fn main() {
    // Collect command-line arguments
    let args: Vec<String> = env::args().collect();
    if args.len() != 2 {
        eprintln!("Usage: {} \"Your message here\"", args[0]);
        std::process::exit(1);
    }

    let message = args[1].as_bytes();

    // Ports to send the message to
    let ports = [8080, 8081, 8082, 8083];

    // Create a UDP socket
    let socket = UdpSocket::bind("0.0.0.0:0").expect("Could not bind socket");

    // Send the message to each port
    for port in &ports {
        let addr = format!("127.0.0.1:{}", port);
        match socket.send_to(message, &addr) {
            Ok(_) => println!("Message sent to port {}", port),
            Err(e) => eprintln!("Failed to send to port {}: {}", port, e),
        }
    }

    // Socket will be closed when it goes out of scope
}
